include/seqan3/alphabet/aminoacid/aminoacid_base.hpp

```diff
@@ -0,0 +1,139 @@
+//     * Redistributions in binary form must reproduce the above copyright
+//       notice, this list of conditions and the following disclaimer in the
+//       documentation and/or other materials provided with the distribution.
+//     * Neither the name of Knut Reinert or the FU Berlin nor the names of
+//       its contributors may be used to endorse or promote products derived
+//       from this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+// ARE DISCLAIMED. IN NO EVENT SHALL KNUT REINERT OR THE FU BERLIN BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+// LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+// OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+// DAMAGE.
+//
+// ============================================================================
+
+/*!\file
+ * \author Hannes Hauswedell <hannes.hauswedell AT fu-berlin.de>
+ * \brief Free function/metafunction wrappers for alphabets with member functions/types.
+ *
+ * This shall not need be included manually, just include `alphabet/concept.hpp`.
+ */
+
+#pragma once
+
+#include <seqan3/alphabet/detail/alphabet_base.hpp>
+#include <seqan3/alphabet/detail/convert.hpp>
+#include <seqan3/alphabet/aminoacid/concept.hpp>
+
+namespace seqan3
+{
+
+/*!\brief A CRTP-base that refines seqan3::alphabet_base and is used by the amino acids.
+ * \tparam derived_type The CRTP parameter type.
+ * \tparam size         The size of the alphabet.
+ * \tparam char_t       The character type of the alphabet (set this to `void` when defining just a
+ *                      seqan3::semi_alphabet_concept).
+ */
+template <typename derived_type, auto size, typename char_t = char>
+class aminoacid_base : public alphabet_base<derived_type, size, char_t>
+{
+private:
+    //!\brief Type of the base class.
+    using base_t = alphabet_base<derived_type, size, char_t>;
+
+    //!\brief Befriend the base class.
+    friend base_t;
+
+    /*!\name Constructors, destructor and assignment
+     * \{
+     */
+    constexpr aminoacid_base() : base_t{} {}
+    constexpr aminoacid_base(aminoacid_base const &) = default;
+    constexpr aminoacid_base(aminoacid_base &&) = default;
+    constexpr aminoacid_base & operator=(aminoacid_base const &) = default;
+    constexpr aminoacid_base & operator=(aminoacid_base &&) = default;
+    //!\}
+
+    //!\brief Befriend the derived class so it can instantiate.
+    friend derived_type;
```

Comment:
hm, in general yes, but in this case befriending the derived type only happens, because the constructors are private so I thought it is better to put it after the constructors...